### PR TITLE
LMS-683 - [API / INTEGRATION] My Course Tab

### PR DESCRIPTION
**Pull request by ragg09-sunasterisk**
## Issue Link
[LMS-683 [API / INTEGRATION] My Course Tab](https://framgiaph.backlog.com/view/LMS-683)	

## Defintion of Done
* [x] should make request API fetch all courses
* [x] sort
* [x] search
* [x] pagination

## Steps to reproduce
- log in as trainee,
- My Course Tab should display data dynamically with search, sort, pagination

### BE
- api/app_sph_lms/api/serializer/course_serializer.py
- api/app_sph_lms/api/urls.py
- api/app_sph_lms/api/view/course_view.py
### FE
- client/src/redux/store.ts
- client/src/sections/dashboard/trainee/CoursesSection.tsx
- client/src/services/traineeAPI.ts
- client/src/shared/components/Card/TraineeCourseCard/index.tsx
- client/src/shared/utils/interface.ts

## Test Cases
_will update soon..._

## Notes
- to test the functionality with data, assign courses to a trainee manually in django admin panel

## Screenshots
My Course Tab

### CODE REVIEW

**Review by Melanthonyando — APPROVED** [review]
### Approved 👍 

Proof :
[Rene Pr.webm](https://github.com/framgia/sph-lms/assets/110366151/903c2100-708e-4388-8622-08175a39a261)

Iteration: 0
Comment: 0
Testing Time : 30 mins.

File: client/src/sections/dashboard/trainee/CoursesSection.tsx [hunk 1]
```diff
@@ -1,39 +1,100 @@
...
 import FilterIcon from '@/src/shared/icons/FilterIcon';
 import ShowIcon from '@/src/shared/icons/ShowIcon';
+import { type CourseCollection } from '@/src/shared/utils';
 import Link from 'next/link';
+import { Fragment, useEffect, useState } from 'react';
 
 const CoursesSection = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
```
Comment (Vali-R) [inline client/src/sections/dashboard/trainee/CoursesSection.tsx:16]: May I know why you are declaring this page again in the components, I mean you already have the page in the slice, which is by the way I do not see you using it

**Review by Vali-R — COMMENTED** [review]
### Bugs
* Click enter on empty search bar [you get no courses to show message]
* Search for a course and try to sort [it does not work]
* Search for a course and double click enter  [you get no courses to show message]
https://github.com/framgia/sph-lms/assets/110364690/c1398c62-21ac-438c-a84a-e54490b6bc95

File: client/src/sections/dashboard/trainee/CoursesSection.tsx [hunk 2]
```diff
@@ -1,38 +1,99 @@
...
+    error,
+    isLoading,
+  } = useGetTraineeCoursesQuery({
+    search,
+    sort: selectedSortOption,
+    pageNumber: page,
+    pageSize: 4,
+  });
+
+  console.log(CourseCollection);
```
Comment (Vali-R) [inline client/src/sections/dashboard/trainee/CoursesSection.tsx:31]:
```suggestion
```

Please remove this

**Review by Vali-R — APPROVED** [review]
Comment: 3
Iteration: 0
Review time: 20mins

**Review by Melanthonyando — APPROVED** [review]
### Approved 👍 

Iteration : 0
Comment : 0
Testing Time : 45 mins.

**Pull request merged**